presto-main/src/test/java/com/facebook/presto/server/TestQueryStateInfoResource.java

```diff
@@ -57,18 +58,30 @@
     public void setup()
     {
         Request request1 = preparePost()
-                .setUri(HttpUriBuilder.uriBuilderFrom(server.getBaseUrl()).replacePath("/v1/statement").build())
+                .setUri(uriBuilderFrom(server.getBaseUrl()).replacePath("/v1/statement").build())
                 .setBodyGenerator(createStaticBodyGenerator("show catalogs", UTF_8))
                 .setHeader(PRESTO_USER, "user1")
                 .build();
         queryResults = client.execute(request1, createJsonResponseHandler(jsonCodec(QueryResults.class)));
 
         Request request2 = preparePost()
-                .setUri(HttpUriBuilder.uriBuilderFrom(server.getBaseUrl()).replacePath("/v1/statement").build())
+                .setUri(uriBuilderFrom(server.getBaseUrl()).replacePath("/v1/statement").build())
                 .setBodyGenerator(createStaticBodyGenerator("show catalogs", UTF_8))
                 .setHeader(PRESTO_USER, "user2")
                 .build();
         client.execute(request2, createJsonResponseHandler(jsonCodec(QueryResults.class)));
+
+        boolean queued = true;
+        while (queued) {
+            queued = false;
+            List<BasicQueryInfo> queryInfos = client.execute(
+                    prepareGet().setUri(uriBuilderFrom(server.getBaseUrl()).replacePath("/v1/query").build()).build(),
+                    createJsonResponseHandler(listJsonCodec(BasicQueryInfo.class)));
+            assertEquals(queryInfos.size(), 2);
+            for (BasicQueryInfo queryInfo : queryInfos) {
+                queued = queued || queryInfo.getState() == QUEUED;
```

Comment:
`queued` is set to `false` at the beginning of the loop, so `queued = queryInfo.getState() == QUEUED` should be enough.